pkg/okteto/config.go

```diff
@@ -27,3 +30,6 @@ func (Config) GetContextCertificate() (*x509.Certificate, error) { return GetCon
 func (Config) IsInsecureSkipTLSVerifyPolicy() bool               { return Context().IsInsecure }
 func (Config) GetServerNameOverride() string                     { return GetServerNameOverride() }
 func (Config) GetContextName() string                            { return Context().Name }
+func (Config) GetExternalRegistryCredentials(registryHost string) (string, string, error) {
+	return GetExternalRegistryCredentials(context.TODO(), registryHost)
```

Comment:
```suggestion
	return GetExternalRegistryCredentials(context.TODO(), registryHost)
```
should we use `context.TODO()` ?

Reply:
This function is mostly executed by internal packages outside of a "context" (registry, credential helpers, etc).

Changing this signature to add a  context would require more more code to account for it and adding `context.TODO()` to the callers.

The `func(registryHost string) (string, string, error)` is a known signatures by all these packages that we need to respect

Reply:
I made some changes to improve this since, at least for buildkit's auth provider, we do have access to the context and we now respect it: https://github.com/okteto/okteto/pull/3799/commits/a537160c89da114b757050b660fc828b5e2102dc